### PR TITLE
Issue 11511 listing table action header

### PR DESCRIPTION
**Pull request by woontsoudev**

### CODE REVIEW

File: src/app/view/components/_common/pattern-library/pattern-library.html [hunk 1]
```diff
@@ -43,6 +43,21 @@ <h3 class="section__title">Data Table</h3>
                     <p-column field="color" header="Color" [sortable]="true"></p-column>
                 </p-dataTable>
             </article>
+            <h3 class="section__title">Action Header</h3>
+            <article>
+                <!--<action-header></action-header>-->
```
Comment (fmontes) [inline src/app/view/components/_common/pattern-library/pattern-library.html]: ?

File: src/app/view/components/_common/pattern-library/pattern-library.ts [hunk 2]
```diff
@@ -9,14 +9,23 @@ import {SelectItem, AutoComplete} from 'primeng/primeng';
 })
 
 export class PatternLibrary {
+    public selectedDummyData = [];
+    public selectedCars = [];
+
+    // tslint:disable-next-line:no-unused-variable
```
Comment (fmontes) [inline src/app/view/components/_common/pattern-library/pattern-library.ts]: Remove all the comments and fix tslints

File: src/app/view/components/_common/pattern-library/pattern-library.ts [hunk 3]
```diff
@@ -91,4 +113,8 @@ export class PatternLibrary {
     showDialog(): void {
         this.displayDialog = true;
     }
+
+    showMessage(): void {
```
Comment (fmontes) [inline src/app/view/components/_common/pattern-library/pattern-library.ts]: Change the name to something more related to the action header.

File: src/app/view/components/_common/action-header/action-header.scss [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        color: $white;
+        display: flex;
+        height: 60px;
+        transform: translateY(60px);
+        visibility: hidden;
+        transition: 
+            visibility 0.5s,
+            transform 0.5s ease;
+
+        span {
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.scss]: add a bem class name.

File: src/app/view/components/_common/action-header/action-header.scss [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            margin-left: 20px;
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.scss]: $basic-padding.

File: src/app/view/components/_common/action-header/action-header.scss [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            padding-right: 20px;
+            border-right: 1px solid white;
+            line-height: 28px;
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.scss]: Use paddings 28px it's too static.

File: src/app/view/components/_common/action-header/action-header.scss [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        transform: translateY(60px);
+        visibility: hidden;
+        transition: 
+            visibility 0.5s,
+            transform 0.5s ease;
+
+        span {
+            margin-left: 20px;
+            padding-right: 20px;
+            border-right: 1px solid white;
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.scss]: $white

File: src/app/view/components/_common/action-header/action-header.scss [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        transform: translateY(-60px);
+        visibility: hidden;
+        transition: 
+            opacity 0.5s ease 0.4s,
+            transform 0.5s ease 0.4s,
+            visibility 0.5s;
+    }
+
+    .action-header__group-actions {
+        // opacity: 1;
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.scss]: Remove all the commented code.

File: src/app/view/components/_common/action-header/action-header.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+    constructor() {
+
+    }
+
+    ngOnChanges(): any {
+        this.dynamicOverflow = 'hidden';
+        setTimeout(() => {
+            this.dynamicOverflow = 'visible';
+        }, 1000);
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.ts]: Please reduce the animation to 500ms 1s it's too slow.

File: src/app/view/components/_common/action-header/action-header.scss [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        opacity: 1;
+        transform: translateY(0);
+        visibility: visible;
+        transition: 
+            visibility 1s linear 0.6s,
+            opacity 0.5s ease 0.3s,
+            transform 0.5s ease 0.5s;
+
+        &:first-child {
+            margin-left: 18px;
```
Comment (fmontes) [inline src/app/view/components/_common/action-header/action-header.scss]: $basic-padding?

File: src/app/view/components/_common/action-header/action-header.html [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<div class="action-header" [ngClass]="{'selected': selected}"  [ngStyle]="{'overflow': dynamicOverflow}">
+    <div class="action-header__global-search">
+        <button pButton actionButton icon="fa-plus" (click)="primaryCommand()"></button>
+        <input pInputText type="text" placeholder="Global search" />
+    </div>
+    <div class="action-header__group-actions">
+        <span class="action-header__selected-items-counter">{{selectedItems.length}} Selected</span>
```
Comment (freddyucv) [inline src/app/view/components/_common/action-header/action-header.html]: dont wee need i18n message here?

File: src/app/view/components/_common/action-header/action-header.spec.ts [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+                label: 'Group Actions 2',
+                model: [
+                    {
+                        command: secondSpy,
+                        icon: 'fa-refresh',
+                        label: 'Action 2-1'
+                    }
+                ]
+            }
+        ];
```
Comment (freddyucv) [inline src/app/view/components/_common/action-header/action-header.spec.ts]: Why happened if just pass one actionButton?

File: src/app/view/components/_common/action-header/action-header.ts [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    templateUrl: 'action-header.html'
+})
+
+export class ActionHeaderComponent {
+
+    public dynamicOverflow = 'visible';
+
+    @Input() selected = false;
+    @Input() selectedItems = [];
+    @Input() actionButtonItems = [];
```
Comment (freddyucv) [inline src/app/view/components/_common/action-header/action-header.ts]: priming splitButton handle the MenuItem interface, so it should be something like:

@Input() actionButtonItems = MenuItem[];

File: src/app/view/components/_common/action-header/action-header.spec.ts (hunk 12 above)
Comment (freddyucv) [inline src/app/view/components/_common/action-header/action-header.spec.ts]: I think that is a good idea do a test with only one button action

File: src/app/view/components/_common/action-header/action-header.ts [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    templateUrl: 'action-header.html'
+})
+
+export class ActionHeaderComponent extends BaseComponent {
+
+    public dynamicOverflow = 'visible';
+
+    @Input() selected = false;
+    @Input() selectedItems = [];
+    @Input() actionButtonItems: MenuItems[];
```
Comment (freddyucv) [inline src/app/view/components/_common/action-header/action-header.ts]: I think that is MenuItem

**Pull request merged**